drivers/SmartThings/zigbee-switch/profiles/inovelli-vzm31-sn.yml

```diff
@@ -0,0 +1,279 @@
+      default: 1
+  - name: "parameter258"
+    title: "258. Switch Mode"
+    description: "Use as a Dimmer or an On/Off switch"
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        "0": "Dimmer"
+        "1": "On/Off (default)"
+      default: 1
+  - name: "parameter22"
+    title: "22. Aux Switch Type"
+    description: "Set the Aux switch type. Smart Bulb Mode does not work in Dumb 3-Way Switch mode."
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        "0": "None (default)"
+        "1": "3-Way Dumb Switch"
+        "2": "3-Way Aux Switch"
+        "3": "Full Sine Wave (fw 2.11+)"
+      default: 0
+  - name: "parameter52"
+    title: "52. Smart Bulb Mode"
+    description: "For use with Smart Bulbs that need constant power and are controlled via commands rather than power. Smart Bulb Mode does not work in Dumb 3-Way Switch mode."
+    required: false
+    preferenceType: enumeration
+    definition:
+      options:
+        "0": "Disabled (default)"
+        "1": "Smart Bulb Mode"
+      default: 0
+  - name: "parameter1"
+    title: "1. Dimming Speed (Remote)"
+    description: "This changes the speed that the light dims up when controlled from the hub. A setting of '0' turns the light immediately on. Increasing the value slows down the transition speed. Value is multiplied by 100ms. 
+  Default=25 (2500ms or 2.5s)"
+    required: false
+    preferenceType: number
+    definition:
+      minimum: 0
+      maximum: 126
+      default: 25
+  - name: "parameter2"
+    title: "2. Dimming Speed (Local)"
+    description: "This changes the speed that the light dims up when controlled at the switch. A setting of '0' turns the light immediately on. Increasing the value slows down the transition speed. Value is multiplied by 100ms. 
+  (i.e 25 = 2500ms or 2.5s) Default=127 (Sync with parameter 1)"
+    required: false
+    preferenceType: number
+    definition:
+      minimum: 0
+      maximum: 127
+      default: 127
+  - name: "parameter3"
+    title: "3. Ramp Rate (Remote)"
+    description: "This changes the speed that the light turns on when controlled from the hub. A setting of '0' turns the light immediately on. Increasing the value slows down the transition speed. Value is multiplied by 100ms. 
+  (i.e 25 = 2500ms or 2.5s) Default=127 (Sync with parameter 1)"
+    required: false
+    preferenceType: number
+    definition:
+      minimum: 0
+      maximum: 127
+      default: 127
+  - name: "parameter4"
+    title: "4. Ramp Rate (Local)"
+    description: "This changes the speed that the light turns on when controlled at the switch. A setting of '0' turns the light immediately on. Increasing the value slows down the transition speed. Value is multiplied by 100ms. 
+  (i.e 25 = 2500ms or 2.5s) Default=127 (Sync with parameter 3)"
+    required: false
+    preferenceType: number
+    definition:
+      minimum: 0
+      maximum: 127
+      default: 127
```

Comment:
One suggestion I would make would just be to unify all 4 of these parameters into 1 that controls the transition speed in all scenarios.

Reply:
We actually have 8 settings for this functionality so I have already consolidated it somewhat. I was told to try to use the same settings as what is in our red series driver that is already in the repo:

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/2125c846a09b5b041f636d82e0855b10689a5859/drivers/SmartThings/zwave-switch/profiles/inovelli-dimmer-power-energy.yml#L48
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/2125c846a09b5b041f636d82e0855b10689a5859/drivers/SmartThings/zwave-switch/profiles/inovelli-dimmer-power-energy.yml#L98
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/2125c846a09b5b041f636d82e0855b10689a5859/drivers/SmartThings/zwave-switch/profiles/inovelli-dimmer-power-energy.yml#L108
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/2125c846a09b5b041f636d82e0855b10689a5859/drivers/SmartThings/zwave-switch/profiles/inovelli-dimmer-power-energy.yml#L118
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/2125c846a09b5b041f636d82e0855b10689a5859/drivers/SmartThings/zwave-switch/profiles/inovelli-dimmer-power-energy.yml#L155